### PR TITLE
Update/docs

### PR DESCRIPTION
**Pull request by h4ck3r-04**
Added Documentation to the Annotation Validator.

### CODE REVIEW

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java [hunk 1]
```diff
@@ -38,39 +38,63 @@ public class ValidationContext {
...
+	 *
+	 * @param env the processing environment
+	 */
 	public ValidationContext(ProcessingEnvironment env) {
 		typeUtils = env.getTypeUtils();
 		elementUtils = env.getElementUtils();
 		messager = env.getMessager();
 
 		LIST_ELEM = elementUtils.getTypeElement(List.class.getCanonicalName());
-		DB_ANNOTATED_OBJECT_ELEM =
```
Comment (nsadeveloper789) [inline Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java:47]: Unintended code formatting?

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java [hunk 2]
```diff
@@ -38,39 +38,63 @@ public class ValidationContext {
...
+	 * @return true if the field has the specified type, false otherwise
+	 */
 	public boolean hasType(VariableElement field, TypeMirror type) {
 		TypeMirror fieldType = field.asType();
 		try {
 			PrimitiveType unboxed = typeUtils.unboxedType(type);
 			if (typeUtils.isSameType(fieldType, unboxed)) {
 				return true;
 			}
-		}
```
Comment (nsadeveloper789) [inline Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java:72]: Incorrect code formatting.

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java [hunk 3]
```diff
@@ -86,9 +110,16 @@ public boolean hasType(VariableElement field, TypeMirror type) {
 		}
 
 		return typeUtils.isAssignable(fieldType, type);
-//		return typeUtils.isSameType(fieldType, type);
```
Comment (nsadeveloper789) [inline Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java:89]: This is okay.

**Review by nsadeveloper789 — CHANGES_REQUESTED** [review]
Thanks for adding these docs. This validator was a bit of an experiment, but Gradle does apply it when building Ghidra, so I suppose it should be documented. That said, please limit your changes to docs, since that's the title of the PR. Your formatter is not configured with our provided settings, and it's mangled our code in some places.

File: .gitignore [hunk 4]
```diff
@@ -91,3 +91,6 @@ gradlew.*
 *.log
 core.*
 !core.png
+
```
[the remark this answers is not in the document]
Comment (h4ck3r-04) [reply .gitignore]: The change has been removed.

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/AbstractDBAnnotationValidator.java [hunk 5]
```diff
@@ -20,24 +20,41 @@
...
+	 * 
+	 * @param annotType the type of the annotation being validated
+	 * @param field the field being validated
+	 * @param type the enclosing type of the field
+	 */
 	protected void checkEnclosingType(Class<? extends Annotation> annotType, VariableElement field,
 			TypeElement type) {
 		if (type.getKind() != ElementKind.CLASS) {
 			ctx.messager.printMessage(Kind.ERROR, String.format(
-				"@%s can only be applied to fields in a class", annotType.getSimpleName()), field);
```
[the remark this answers is not in the document]
Comment (h4ck3r-04) [reply Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/AbstractDBAnnotationValidator.java]: Reverted the code to initial format.

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/AccessSpec.java [hunk 6]
```diff
@@ -19,11 +19,19 @@
...
+/**
+ * The {@code AccessSpec} enum represents different levels of access specifiers
+ * (private, package-private, protected, public) with corresponding access levels
+ */
 public enum AccessSpec {
 	PRIVATE(0), PACKAGE(1), PROTECTED(2), PUBLIC(3);
 
 	private final int level;
 
+	/**
```
[the remark this answers is not in the document]
Comment (h4ck3r-04) [reply Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/AccessSpec.java]: Removed the documentation for the private constructor.

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java (hunk 1 above)
Comment (h4ck3r-04) [reply Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java]: Reverted to the old code formatting

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java (hunk 2 above)
Comment (h4ck3r-04) [reply Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/ValidationContext.java]: Fixed

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/DBAnnotatedFieldValidator.java [hunk 7]
```diff
@@ -39,6 +45,11 @@ public class DBAnnotatedFieldValidator extends AbstractDBAnnotationValidator {
 
 	final TypeElement ENUM_CODEC_ELEM;
 
+	/**
+	 * Constructs a new {@code DBAnnotatedFieldValidator} with the specified validation context and field element.
```
Comment (nsadeveloper789) [inline Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/DBAnnotatedFieldValidator.java:49]: "Construct" (no s)

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/AccessSpec.java [hunk 8]
```diff
@@ -19,6 +19,10 @@
 
 import javax.lang.model.element.Modifier;
 
+/**
+ * The {@code AccessSpec} enum represents different levels of access specifiers
```
Comment (nsadeveloper789) [inline Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/AccessSpec.java:23]: "An enum to represent ..."

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/AbstractDBAnnotationValidator.java [hunk 9]
```diff
@@ -20,24 +20,41 @@
 import javax.lang.model.element.*;
 import javax.tools.Diagnostic.Kind;
 
+/**
```
[the remark this answers is not in the document]
Comment (nsadeveloper789) [reply Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/AbstractDBAnnotationValidator.java:23]: Not sure I was clear enough here. I'll use this as an example (and I'll do another example below).

```java
/**
 * An abstract class for validation annotations on {@link DBAnotatedObject}.
 * 
 * <p>
 * Performs validation checks on annotated fields and their enclosing types.
 */
```

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/DBAnnotatedColumnValidator.java [hunk 10]
```diff
@@ -22,14 +22,34 @@
...
+	 * Construct a new {@code DBAnnotatedColumnValidator} with the specified validation context and the column element.
+	 * @param ctx
+	 * @param column
+	 */
 	public DBAnnotatedColumnValidator(ValidationContext ctx, VariableElement column) {
 		super(ctx);
 		this.column = column;
 	}
 
+	/**
```
Comment (nsadeveloper789) [inline Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/DBAnnotatedColumnValidator.java:43]: Another example where I'd like to see details separated from the brief:

```java
/**
 * Validate the annotated column field.
 * 
 * <p>
 * It performs the following checks to ensure the field meets the requirements for a database column:
 * <ul>
 * <li>...</li>
 * ...
 * </ul>
 */

File: Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/DBAnnotatedFieldValidator.java [hunk 11]
```diff
@@ -101,13 +143,25 @@ public void validate() {
 		checkCodecTypes(type);
 	}
 
+	/**
+	 * Returns the default codec type element for the specified Java type.
```
Comment (nsadeveloper789) [inline Ghidra/Debug/AnnotationValidator/src/main/java/ghidra/util/database/annotproc/DBAnnotatedFieldValidator.java:147]: "Return" (no s)

**Review by nsadeveloper789 — CHANGES_REQUESTED** [review]
Thanks for reverting the code formatting. I don't mean to be annoying, but you missed a few spots. I've commented on a few and provided (I hope clearer) examples.

**Review by nsadeveloper789 — APPROVED** [review]
Awesome. Thanks for putting up with my pedantry in the javadocs. There are a few tweaks left, but it's pretty much just me running my doc formatter, so I can do that on our end.

**Pull request closed without merging**